yellowbrick/features/projection.py

```diff
@@ -172,26 +184,36 @@ def ax(self):
     def ax(self, ax):
         self._ax = ax
 
-    def layout(self):
+    def layout(self, divider=None):
         """
         Creates the layout for colorbar when target type is continuous.
         The colorbar is added to the right of the scatterplot.
 
         Subclasses can override this method to add other axes or layouts.
         """
-        if self._cax is None and self._target_color_type == TargetType.CONTINUOUS:
-            # Ensure matplotlib version compatibility
-            if make_axes_locatable is None:
-                raise YellowbrickValueError((
-                    "Colorbar requires matplotlib 2.0.2 or greater "
-                    "please upgrade matplotlib"
-                ))
-
-            # Create the new axes for the colorbar
-            divider = make_axes_locatable(self.ax)
-            self._cax = divider.append_axes("right", size="5%", pad=0.3)
-            self._cax.set_yticks([])
-            self._cax.set_xticks([])
+        if (
+            self._cax is None
+            and self._target_color_type == TargetType.CONTINUOUS
+            and self.colorbar
+        ):
+            if self.projection == 2:
```

Comment:
What happens if the projection is 3? Can we not do a colorbar (it seemed like I saw one in the tests)? If you're just using `plt.colobar` in the 3D case then you could probably move this condition to the if above, e.g.

```python
        if (
            self._target_color_type == TargetType.CONTINUOUS
            and self.projection == 2
            and self.colorbar
            and self._cax is None
        ):
```

Note that I like to order if statements by the most likely to be false first in order to more easily short circuit the if.

Reply:
In the case of 3d projections, we do have a colorbar when the target is continuous. I will move this condition above!